Classes/Service/FolderAccessService.php

```diff
@@ -59,34 +50,29 @@ public function checkResourceAccess(ResourceStorage $storage, string $identifier
         if ($this->context->getPropertyFromAspect('backend.user', 'isLoggedIn')) {
             return true;
         }
+
         /** @var UserAspect $feUser */
         $feUser = $this->context->getAspect('frontend.user');
         // early exit, if we don't have a frontend login
         if (!$feUser->isLoggedIn()) {
             return false;
         }
 
-
         // detect current users groups and
         // folder access groups, if given
         // otherwise look into file storage
         $datasetGroupArray = $feUser->getGroupIds();
         $folderAccess = $this->findDirectoryAccess($storage, $identifier);
         $storageRecord = $storage->getStorageRecord();
-        $groupArray = $folderAccess
+        $groupArray = $folderAccess instanceof \Calien\SecureFilemount\Domain\Model\Folder
```

Comment:
FQN should be simplified here, using `use` namespace import and short name (eventually as alias)